_posts/2020-02-07-metal3-dev-env-install-deep-dive.md

```diff
@@ -0,0 +1,921 @@
+```
+
+Below are detailed the containers inside the *infra-pod* pod which are running as privileged using the host network:
+<br>
+<br>
+> * **The httpd container.** 
+<br>
+<br>
+> A folder called *shared* where the cloud OS image and IPA files are available is mounted and exposed to the virtual bare metal hosts. 
+<br>
+
+>```sh
++ sudo podman run -d --net host --privileged --name httpd-infra --pod infra-pod -v /opt/metal3-dev-env/ironic:/shared --entrypoint /bin/runhttpd quay.io/metal3-io/ironic
+```
+<br>
+> This folder also contains the `inspector.ipxe` file which contains the information needed to be able to run the `ironic-python-agent` kernel and initramfs. Below, httpd-infra container is accessed and it has been verified that host's `/opt/metal3-dev-env/ironic/` (`IRONIC_DATA_DIR`) is mounted inside the *shared* folder of the container:
+
+```sh
+[alosadag@eko1 metal3-dev-env]$ sudo podman exec -it httpd-infra bash
+[root@infra-pod shared]# cat html/inspector.ipxe 
+#!ipxe
+
+:retry_boot
+echo In inspector.ipxe
+imgfree
+# NOTE(dtantsur): keep inspection kernel params in [mdns]params in ironic-inspector-image
+kernel --timeout 60000 http://172.22.0.1:80/images/ironic-python-agent.kernel ipa-inspection-callback-url=http://172.22.0.1:5050/v1/continue ipa-inspection-collectors=default,extra-hardware,logs systemd.journald.forward_to_console=yes BOOTIF=${mac} ipa-debug=1 ipa-inspection-dhcp-all-interfaces=1 ipa-collect-lldp=1 initrd=ironic-python-agent.initramfs || goto retry_boot
+initrd --timeout 60000 http://172.22.0.1:80/images/ironic-python-agent.initramfs || goto retry_boot
+boot
+```
+<br>
+<br>
+> * **The vbmc container.**
+<br>
+<br>
+> This container mounts two host folders. One is the `/opt/metal3-dev-env/virtualbmc/vbmc` where the `vbmc` configuration for each node is stored. The other folder is the `/root/.ssh` where root keys are located, specifically `id_rsa_virt_power` which is used to manage the communication with `libvirt`.
+
+> ```sh
+> + sudo podman run -d --net host --privileged --name vbmc --pod infra-pod -v /opt/metal3-dev-env/virtualbmc/vbmc:/root/> .vbmc -v /root/.ssh:/root/ssh quay.io/metal3-io/vbmc
+> ```
+<br>
+<br>
+> * **The sushy-tools container.** 
+<br>
+<br>
+> This container mounts the `/opt/metal3-dev-env/virtualbmc/sushy-tools config folder and the `/root/.ssh` local folder as well. The functionality is similar as the `vbmc`, however this use redfish instead of ipmi to connect to the BMC.
+
+>```sh
+> + sudo podman run -d --net host --privileged --name sushy-tools --pod infra-pod -v /opt/metal3-dev-env/virtualbmc/> sushy-tools:/root/sushy -v /root/.ssh:/root/ssh quay.io/metal3-io/sushy-tools
+> ```
+
+<br>
+
+> **Information:** At this point the virtual infrastructure must be ready to apply the Kubernetes specific configuration. Note that all VMs specified by `NUMBER_NODES` and `minikube` must be shut down and the defined virtual network must be active:
```

Comment:
Note that all **the** VMs specified by NUMBER_NODES and minikube must be shut down and the defined virtual network must be active: